app/controllers/application_controller.rb

```diff
@@ -2,4 +2,12 @@
 
 class ApplicationController < ActionController::API
   include DeviseTokenAuth::Concerns::SetUserByToken
+  before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :authenticate_user!
```

Comment:
I'm not sure and I'm just thinking but isn't it make all endpoints private? I mean you do not have to be signed in to see movies but you have to be signed in to buy tickets

Reply:
Since there will be probably more controllers where I need authentication than those where I don't need it, my idea was to ignore ('skip') authentication for specific controllers. :)

Reply:
You can add `skip_before_action :authenticate_user!` to have some routes public, so there are two cases when doing it that way is better:
- You have way more private routes than public and it is just better to default to private and make the few public accessible  without authorization
- You want to default settings be the most restrictive settings for security reasons